File: qe-selenium-python/Activity11.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-controls")
    wait = WebDriverWait(driver, 20)
    print(driver.title)
    actions = webdriver.ActionChains(driver)
    
    print("Title of the Page: "+driver.title)

    check_box = driver.find_element(By.ID, "checkbox")
    toggle_button = driver.find_element(By.CSS_SELECTOR, ("section.mx-auto:nth-child(1) > button:nth-child(2)"))

    actions.click(toggle_button).perform()
    actions.click(toggle_button).perform()

    wait.until(EC.visibility_of(check_box))

    actions.click(check_box).perform()

File: qe-selenium-java/Activity11.java
package demos;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity11 {
public static void main(String[] args) {
    WebDriver driver = new FirefoxDriver();
    WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(10));
    driver.get("https://training-support.net/webelements/dynamic-controls");
    System.out.println("Title of the Page: " + driver.getTitle());

    WebElement checkBox = driver.findElement(By.id("checkbox"));
    WebElement toggleButton = driver.findElement(By.cssSelector("section.mx-auto:nth-child(1) > button:nth-child(2)"));
    
    Actions actions = new Actions(driver);

    actions.click(toggleButton).perform();
    actions.click(toggleButton).perform();

    wait.until(ExpectedConditions.visibilityOf(checkBox));

    actions.click(checkBox).perform();

    driver.quit();


}
}
